core/src/odbc_uri.rs

```diff
@@ -320,28 +326,53 @@ impl ODBCUri {
         let pwd = self.remove_mandatory_attribute(PWD_KWS)?;
         let server = self.remove_mandatory_attribute(SERVER_KWS)?;
         let cred = Credential::builder().username(user).password(pwd).build();
+        let app_name = self.handle_app_name(None);
+        let driver_name = self.handle_driver_info(app_name.as_ref().unwrap());
         Ok(UserOptions {
             client_options: ClientOptions::builder()
                 .hosts(vec![
                     ServerAddress::parse(server).map_err(Error::InvalidClientOptions)?
                 ])
                 .credential(cred)
-                .app_name(self.handle_app_name())
+                .app_name(app_name)
+                .driver_info(driver_name)
                 .build(),
             uuid_representation: None,
         })
     }
 
-    fn handle_app_name(&mut self) -> Option<String> {
-        self.remove(&[APPNAME])
-            .map(|s| {
-                if s == POWERBI_CONNECTOR {
-                    format!("{}+{}", POWERBI_CONNECTOR, DRIVER_METRICS_VERSION.as_str())
+    fn handle_driver_info(&self, app_name: &'_ str) -> DriverInfo {
+        let driver_info = DriverInfo::builder();
+        let driver_name = if app_name.contains(POWERBI_CONNECTOR) {
+            format!("{}|{}", DRIVER_SHORT_NAME, POWERBI_CONNECTOR)
+        } else {
+            DRIVER_SHORT_NAME.to_string()
+        };
+        driver_info.name(driver_name).build()
+    }
+
+    fn handle_app_name(&mut self, app_name: Option<String>) -> Option<String> {
+        if let Some(ref app_name) = app_name {
+            if app_name.contains(POWERBI_CONNECTOR) {
+                log::info!("Connecting with PowerBI Connector version: {}", app_name);
+            }
+        }
```

Comment:
This would handle [SQL-1565](https://jira.mongodb.org/browse/SQL-1565). Up to reviewers if we should do this now or as a separate ticket.